### PR TITLE
Fixed current location panning out of bounds

### PR DESCRIPTION
**Pull request by bryant100000**
Connects to #1357

### CODE REVIEW

File: opentreemap/treemap/js/src/mapPage/addMapFeature.js [hunk 1]
```diff
@@ -185,8 +192,16 @@ function init(options) {
...
     function onGeolocateSuccess(lonLat) {
-        var latLng = L.latLng(lonLat.coords.latitude, lonLat.coords.longitude);
-        onLocationChosen(latLng);
+	if (!pointInMap(lonLat)) {
+	    $pointInStreamError.show();
+	} else {
+            var latLng = L.latLng(lonLat.coords.latitude, lonLat.coords.longitude);
+            onLocationChosen(latLng);
+        }
+    }
```
Comment (maurizi) [inline opentreemap/treemap/js/src/mapPage/addMapFeature.js]: The indentation in this function is a little off.

File: opentreemap/treemap/js/src/mapPage/addMapFeature.js [hunk 2]
```diff
@@ -40,13 +41,15 @@ function init(options) {
...
+        $pointInStreamError = U.$find('.pointnotinmap-error', $sidebar),
         triggerSearchBus = options.triggerSearchBus,
 
         $form = U.$find(formSelector, $sidebar),
         editFields = formSelector + ' [data-class="edit"]',
         validationFields = options.validationFields || formSelector + ' [data-class="error"]',
         $placeMarkerMessage = U.$find('.place-marker-message', $sidebar),
-        $moveMarkerMessage = U.$find('.move-marker-message', $sidebar);
+        $moveMarkerMessage = U.$find('.move-marker-message', $sidebar),
+	boundsGeoJson = L.geoJson(config.instance.bounds);
```
Comment (maurizi) [inline opentreemap/treemap/js/src/mapPage/addMapFeature.js]: indentation is off here too.

File: opentreemap/treemap/js/src/mapPage/addMapFeature.js [hunk 3]
```diff
@@ -185,8 +192,16 @@ function init(options) {
...
+	if (!pointInMap(lonLat)) {
+            $pointInStreamError.show();
+	} else {
+            var latLng = L.latLng(lonLat.coords.latitude, lonLat.coords.longitude);
+            onLocationChosen(latLng);
+        }
+    }
+
+    function pointInMap(lonLat) {
+        var polysForPoint = leafletPip.pointInLayer(lonLat, boundsGeoJson, true);
```
[the remark this answers is not in the document]
Comment (maurizi) [reply opentreemap/treemap/js/src/mapPage/addMapFeature.js]: @jwalgran The following line should fill in this function
```javascript
return polysForPoint.length > 0;
```

**Comment by jwalgran** [comment]
Thank you for the contribution. I extended this PR and made some fixes in #3242.

**Pull request closed without merging**